File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/entities/Grade.java
package mx.edu.upqroo.kristenandroid.data.database.entities;

import com.google.gson.annotations.SerializedName;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

import static androidx.room.ForeignKey.CASCADE;

@Entity(tableName = "grade",
        indices = {@Index("userId")},
        foreignKeys = @ForeignKey(entity = UserInformation.class,
                parentColumns = "userId",
                childColumns = "userId",
                onDelete = CASCADE))
public class Grade {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "gradeId")
    private long gradeId;

    @NonNull
    @SerializedName("grupo")
    @ColumnInfo(name = "code")
    private String code;

    @NonNull
    @SerializedName("nombre_mat")
    @ColumnInfo(name = "subject")
    private String subject;

    @NonNull
    @SerializedName("calificacion")
    @ColumnInfo(name = "generalGrade")
    private String generalGrade;

    @NonNull
    @SerializedName("parcial_1")
    @ColumnInfo(name = "gradeOne")
    private String gradeOne;

    @NonNull
    @SerializedName("parcial_2")
    @ColumnInfo(name = "gradeTwo")
    private String gradeTwo;

    @NonNull
    @SerializedName("parcial_3")
    @ColumnInfo(name = "gradeThree")
    private String gradeThree;

    @NonNull
    @SerializedName("parcial_4")
    @ColumnInfo(name = "gradeFour")
    private String gradeFour;

    @NonNull
    @SerializedName("parcial_5")
    @ColumnInfo(name = "gradeFive")
    private String gradeFive;

    @NonNull
    @SerializedName("matricula")
    @ColumnInfo(name = "userId")
    private String userId;

    public Grade(long gradeId,
                 @NonNull String code,
                 @NonNull String subject,
                 @NonNull String generalGrade,
                 @NonNull String gradeOne,
                 @NonNull String gradeTwo,
                 @NonNull String gradeThree,
                 @NonNull String gradeFour,
                 @NonNull String gradeFive,
                 @NonNull String userId) {
        this.gradeId = gradeId;
        this.code = code;
        this.subject = subject;
        this.generalGrade = generalGrade;
        this.gradeOne = gradeOne;
        this.gradeTwo = gradeTwo;
        this.gradeThree = gradeThree;
        this.gradeFour = gradeFour;
        this.gradeFive = gradeFive;
        this.userId = userId;
    }

    public long getGradeId() {
        return gradeId;
    }

    @NonNull
    public String getCode() {
        return code;
    }

    @NonNull
    public String getSubject() {
        return subject;
    }

    @NonNull
    public String getGeneralGrade() {
        return generalGrade;
    }

    @NonNull
    public String getGradeOne() {
        return gradeOne;
    }

    @NonNull
    public String getGradeTwo() {
        return gradeTwo;
    }

    @NonNull
    public String getGradeThree() {
        return gradeThree;
    }

    @NonNull
    public String getGradeFour() {
        return gradeFour;
    }

    @NonNull
    public String getGradeFive() {
        return gradeFive;
    }

    @NonNull
    public String getUserId() {
        return userId;
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/entities/Grade.kt
package mx.edu.upqroo.kristenandroid.data.database.entities

import androidx.room.*
import androidx.room.ForeignKey.CASCADE
import com.google.gson.annotations.SerializedName

@Entity(tableName = "grade",
        indices = [Index("userId")],
        foreignKeys = [ForeignKey(entity = UserInformation::class,
                parentColumns = ["userId"],
                childColumns = ["userId"],
                onDelete = CASCADE)])
class Grade(@field:PrimaryKey(autoGenerate = true)
            @field:ColumnInfo(name = "gradeId")
                val gradeId: Long,
            @field:SerializedName("grupo")
            @field:ColumnInfo(name = "code")
                val code: String,
            @field:SerializedName("nombre_mat")
            @field:ColumnInfo(name = "subject")
                val subject: String,
            @field:SerializedName("calificacion")
            @field:ColumnInfo(name = "generalGrade")
                val generalGrade: String,
            @field:SerializedName("parcial_1")
            @field:ColumnInfo(name = "gradeOne")
                val gradeOne: String,
            @field:SerializedName("parcial_2")
            @field:ColumnInfo(name = "gradeTwo")
                val gradeTwo: String,
            @field:SerializedName("parcial_3")
            @field:ColumnInfo(name = "gradeThree")
                val gradeThree: String,
            @field:SerializedName("parcial_4")
            @field:ColumnInfo(name = "gradeFour")
                val gradeFour: String,
            @field:SerializedName("parcial_5")
            @field:ColumnInfo(name = "gradeFive")
                val gradeFive: String,
            @field:SerializedName("matricula")
            @field:ColumnInfo(name = "userId")
                val userId: String)
